sdks/react/src/hooks/useDeploy.tsx

```diff
@@ -11,9 +11,9 @@ export function useDeploy() {
   const { mutateAsync, ...mutation } = useMutation<
     Awaited<ReturnType<typeof sdk.deploy>>,
     unknown,
-    Parameters<typeof sdk.deploy>[0]
+    Parameters<typeof sdk.deploy>
   >(data => {
-    return sdk.deploy(data);
+    return sdk.deploy(...data);
```

Comment:
I could be getting confused here, but the type of `data` in this instance is actually:

```ts
deploy(nft: NFTMetadata, transactionArgs?: Transaction)
```

However, the `mutateAsync` just thinks its going to receive `NFTMetadata`. I think its something to do with this:

```ts
Parameters<typeof sdk.deploy>[0]
```

![Screenshot 2022-07-27 at 14 05 39](https://user-images.githubusercontent.com/996430/181254157-ed4bf1f7-29c3-4fa5-8d68-26a6d32577dd.png)

Reply:
The mutation functions from React Query can only take one argument really.

Given the current typing you'd have to do this to call deploy.

```tsx
deploy([{
  maxSupply: 100,
  mintingPrice: 0.01,
  name: 'Test',
  symbol: 'TEST',
  url: 'ipfs://',
}]);
```

Given that there's now two sets of arguments, one being pretty dynamic and the other being just static config. It may be better to handle it like...

```tsx
const { deploy } = useDeploy({ maxFeePerGas: 100 })
deploy({
  maxSupply: 100,
  mintingPrice: 0.01,
  name: 'Test',
  symbol: 'TEST',
  url: 'ipfs://',
})
```

Reply:
@benjaminreid Hopefully, this should now be updated to match that use case you documented ☝️